### PR TITLE
DIV-7689 - Accounting Locked Period details - adding in Preferences from QB API

### PR DESCRIPTION
**Pull request by maniakjaniak**
https://divipay.atlassian.net/browse/DIV-7689

Added in Preferences reference from the QB API
New test for Preferences.
Fixed deprecated assertEquals to assertEqual in test suite. 

Relevant files for "Preferences" change:
CHANGELOG.rst
quickbooks/client.py
quickbooks/objects/__init__.py
quickbooks/objects/preferences.py
tests/unit/objects/test_preferences.py

### CODE REVIEW

File: tests/unit/objects/test_preferences.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+def check_valid_date(date_text):
+    try:
+      datetime.datetime.strptime(date_text, '%Y-%m-%d')
+      return True
+    except ValueError:
+      raise ValueError("Incorrect data format, should be YYYY-MM-DD")
+
+
+class PreferencesTests(unittest.TestCase):
```
Comment (matt-quantblack) [inline tests/unit/objects/test_preferences.py]: Are we using pytest functional tests going forward instead of unittest.TestCase class based tests?

Comment (maniakjaniak) [reply tests/unit/objects/test_preferences.py]: Yeah I'd like to. Not sure if I should make that change in this PR or commit to move them all over in another?

I'll edit my new test to Pytest and we can add the tech-debt / move to the backlog.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,5 +1,8 @@
 Changelog
 =========
+* 0.8.2 (March 7th, 2022)
```
Comment (mattjegan) [inline CHANGELOG.rst:3]: You may come up against the original maintainers when you try get this merged back into trunk but it's fine for our branch

**Review by mattjegan — APPROVED** [review]

File: CHANGELOG.rst (hunk 2 above)
Comment (maniakjaniak) [reply CHANGELOG.rst:3]: Yeah I wasn't planning on opening a PR on the original package. Figured keeping to our forked version would be ok.

Thanks :)

**Pull request merged**